Integrations/integration-DemistoRESTAPI.yml

```diff
@@ -0,0 +1,98 @@
+description: Use Demisto server REST APIs
+configuration:
+- display: Demisto Server URL
+  name: url
+  defaultvalue: ""
+  type: 0
+  required: true
+- display: Demisto Server API Key
+  name: apikey
+  defaultvalue: ""
+  type: 4
+  required: true
+- display: Trust certificate (insecure)
+  name: insecure
+  defaultvalue: "false"
+  type: 8
+  required: false
+- display: User system proxy settings
+  name: proxy
+  defaultvalue: "true"
+  type: 8
+  required: false
+script:
+  script: |
+    var serverURL = params.url;
+    if (serverURL.slice(-1) === '/') {
+        serverURL = serverURL.slice(0,-1);
+    }
+
+    var sendRequest = function(method, url, body) {
+        var requestUrl = serverURL;
+        if (url.slice(-1) != '/') {
+            requestUrl += '/';
+        }
+        requestUrl += url;
+
+        var res = http(
+            requestUrl,
+            {
+                Method: method,
+                Headers: {
+                    'Accept': ['application/json'],
+                    'content-type': ['application/json'],
+                    'authorization': [params.apikey]
+                },
+                Body: body
+            },
+            params.insecure,
+            params.proxy
+            );
+
+        if (res.StatusCode < 200 || res.StatusCode >= 300) {
+            throw 'Demisto REST APIs - Request Failed.\nStatus code: ' + res.StatusCode + '.\nBody: ' + JSON.stringify(res) + '.';
+        }
+
+        try {
+            var json = JSON.parse(res.Body);
+            return {response:json};
+        } catch (ex) {
+            throw 'Demisto REST APIs - Error parsing response - ' + ex;
+        }
+    };
+
+    switch (command) {
+        case 'test-module':
+            sendRequest('GET','system/config');
+            return 'ok';
+        case 'demisto-api-post':
+            var body = JSON.parse(args.body);
+            return sendRequest('POST',args.url, args.body);
+        case 'demisto-api-get':
```

Comment:
I would implement also PUT and DELETE

Reply:
Almost all delete actions require the DELETE method.. delete job, delete cred, delete list, delete task, delete report, delete api key, delete classifier, delete account, delete role, delete integration, delete incident type, delete reputation, delete incident field and many many more.
Same for PUT (which is update action...)

Reply:
OK. So DELETE do not require body, and PUT does?

Reply:
Correct.

Reply:
Basically you implemented the 'http' command again :)